glue/external/echo/selection.py

```diff
@@ -26,8 +26,9 @@ def __init__(self, default_index=0, choices=None, display_func=None, **kwargs):
         self._force_next_sync = WeakKeyDictionary()
 
     def __set__(self, instance, value):
-        if value is not None and value not in self.get_choices(instance):
-            raise ValueError('value {0} is not in valid choices'.format(value))
+        choices = self.get_choices(instance)
+        if value is not None and value not in choices:
+            raise ValueError('value {0} is not in valid choices: {}'.format(value, choices))
```

Comment:
You can't mix numbered and un-numbered formatting - can you put ``{1}`` instead? (just to be more explicit)